test/signalfx_test.js

```diff
@@ -2,20 +2,24 @@ const assume = require('assume');
 const load = require('../src/main');
 const SignalFxRest = require('../src/signalfx-rest');
 
-suite('SignalFxRest', () => {
+suite('SignalFxRest', function() {
   let rest;
+  let apiTokenPresent = false;
 
   suiteSetup(async function() {
     const cfg = await load('cfg', {profile: 'test'});
-    if (!cfg.signalfx.apiToken) {
-      this.skip();
+    if (cfg.signalfx.apiToken) {
+      apiTokenPresent = true;
+      rest = new SignalFxRest(cfg.signalfx.apiToken);
     }
-
-    rest = new SignalFxRest(cfg.signalfx.apiToken);
   });
 
-  suite('timeserieswindow', async () => {
-    test('throws an error for a nonexistent metric', async () => {
+  suite('timeserieswindow', async function() {
+    test('throws an error for a nonexistent metric', async function() {
+      if (!apiTokenPresent) {
+        this.skip();
+      }
```

Comment:
:+1:

Reply:
You can also write this once inside a `setup` at the suite level:
```javascript
suite('timeserieswindow', async function() {
  setup(function() {
    if (!apiTokenPresent) {
      this.skip();
    }
  });
```
This way you don't need to include the if statement in both tests.

Reply:
Are setup() and suiteSetup() the same?
Because this.skip() didn't work in suiteSetup()

Reply:
setup runs once for each test, and suiteSetup only once for the suite.  My memory is that this.skip only works for tests, so only in setup.